src/coreclr/vm/dllimport.cpp

```diff
@@ -4291,7 +4290,13 @@ static void CreateNDirectStubAccessMetadata(
         // P/Invoke marked with UnmanagedCallersOnlyAttribute is not
         // presently supported.
         if (pMD->HasUnmanagedCallersOnlyAttribute())
-            COMPlusThrow(kNotSupportedException, IDS_EE_NDIRECT_UNSUPPORTED_SIG);
+        {
+            SString namespaceOrClassName;
+            SString methodName;
+            pMD->GetMethodInfoNoSig(namespaceOrClassName, methodName);
+            COMPlusThrow(kNotSupportedException, IDS_EE_NDIRECT_NAMED_UNSUPPORTED_SIG,
```

Comment:
`UnmanagedCallersOnlyAttribute` is not part of the method signature. It is an attribute on a method.

Would it be better if the message just said that UnmanagedCallersOnlyAttribute is not supported for PInvoke methods?

Reply:
Sure, I can do that.